addons/account/models/account_move_line.py

```diff
@@ -654,6 +654,8 @@ def _compute_amount_currency(self):
                 line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
             if line.currency_id == line.company_id.currency_id:
                 line.amount_currency = line.balance
+            elif line.amount_currency == line.balance:
+                line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
```

Comment:
It would be more interesting to see why it is equal instead of changing it if it is

Reply:
It seems expected, we start with an invoice in foreign currency:
* move currency is set to company currency
  * change is propagated to lines currency
  * amount currency is set equal to balance
* move currency is set to foreign currency again
  * change is propagated to lines currency
  * currency rate is updated
  * now we are in that `elif` case, amount currency is equal to balance but currency_id is not company currency
What should be the expected behavior?